w3af_batch

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+
+import logging
+import argparse
+import signal
+import threading as th
+import multiprocessing as mp
+import multiprocessing.dummy as mpd
+
+
+STOPSCAN = mp.Event()
+logger = logging.getLogger('w3af_batch')
+
+
+def cancel_scan(*args):
+    """
+    Dummy SIGINT hanlder which stops target list processing.
+    """
+    STOPSCAN.set()
+    raise KeyboardInterrupt('Scan is terminated by user request')
+
+
+def execute_scan(target, profile, timeout):
+    """
+    Execute w4af scan.  Try to stop it after timeout seconds.
+    """
+    def stop_scan(scanner):
+        """
+        Stop w3af scan after timeout.
+        """
+        if scanner.status.is_running():
+            scanner.stop()
+            logger.info('Scan of target %s is stopped', target)
+
+    from w3af.core.controllers.w3afCore import w3afCore
+    from w3af.core.data.kb.knowledge_base import kb
+
+    scanner = w3afCore()
+    scanner.profiles.use_profile(profile)
+    option_list = scanner.target.get_options()
+    option_list['target'].set_value(target)
+    scanner.target.set_options(option_list)
+    scanner.plugins.init_plugins()
+
+    timer = th.Timer(timeout, stop_scan, (scanner,))
+    timer.start()
+    scanner.start()
+    if timer.is_alive():
+        timer.cancel()
+        timer.join()
+        logger.debug('Target %s has %s vulnerabilities',
+                     target, len(kb.get_all_vulns()))
+
+
+def start_executor(target, profile, timeout):
+    """
+    Start w3af scan in a separate process. Terminate child process after
+    2 * timeout seconds.
```

Comment:
It should act as a precaution. I want to stop long-running scans with w3afCore.stop(). Maybe it would be better to add some additional timeout and then terminate worker process. What do you think?